src/ipa_pc/mod.rs

```diff
@@ -968,12 +973,18 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         let mut points = vec![];
 
         for (_point_label, (point, labels)) in query_to_labels_map.into_iter() {
-            for label in labels.into_iter() {
+            for &label in labels.iter() {
```

Comment:
You consume query_to_labels_map, so let's consume also the labels at this point, it was okey like before (you can save a label.clone() in v_i I think)

Reply:
In actual implementation in "dlog_segmentation" branch "query_to_labels_map" was usage was removed